ui/src/app/shared/services/requests.ts

```
@@ -68,7 +68,18 @@ export default {
 
     loadEventSource(url: string): Observable<string> {
         return Observable.create((observer: Observer<any>) => {
-            let eventSource = new EventSource(`${apiRoot()}${url}`);
+            const fullUrl = `${apiRoot()}${url}`;
+
+            // If there is an error, show it beforehand
+            fetch(fullUrl).then(response => {
+                if (!response.ok) {
+                    return response.text().then(text => {
+                        observer.error({status: response.status, statusText: response.statusText, body: text});
+                    });
+                }
+            });
```

Is the preflight request necessary? Wouldn't the `onerror` handler take care of this anyway?

---

@crenshaw-dev The preflight request is needed because the onerror handler does not handle the HTTP request error (400 Bad Request) and swallows the error message.
Instead EventSource considers it a connection problem and throws the generic error`('connection got closed unexpectedly')`